### PR TITLE
Global delay factor

### PR DESCRIPTION
**Pull request by FragmentedPacket**
This is to add support for global_delay_factor option in the ntc_file_copy module for Ansible.

This adds the global_delay_factor key value as an option and adds it to **kwargs dictionary that is then passed onto pyntc modules (there is an additional pull request over there to make this possible)

I ran into an issue with verifying the MD5 calculation taking too long on the slavebootflash: of a Cisco Catalyst 4500, which this fixes. The values I have tested to fix it is 10, 15, and 300. 300 Timed out and created a socket.error, but I'm not sure if that was due to the setting or just a connection issue I had at that time.

### CODE REVIEW

**Comment by FragmentedPacket** [comment]
The pyntc pull request can be found here: https://github.com/networktocode/pyntc/pull/24

**Comment by itdependsnetworks** [comment]
Can you remove the chmod's that were done?

File: library/ntc_file_copy.py [hunk 1]
```diff
@@ -191,6 +194,7 @@ def main():
             secret=dict(required=False, no_log=True),
             transport=dict(required=False, choices=['http', 'https']),
             port=dict(required=False, type='int'),
+	    global_delay_factor=dict(default=1, required=False),
```
Comment (itdependsnetworks) [inline library/ntc_file_copy.py]: Seems to be a tab here, can you change to whitespace.

**Comment by FragmentedPacket** [comment]
I believe everything is fixed for this now.

**Comment by FragmentedPacket** [comment]
`  tasks:
    - name: "1a. Test with an integer"
      ntc_file_copy:
        platform: cisco_ios_ssh
        local_file: /home/ios-images/4500/{{ approved_version_file }}
        remote_file: "{{ approved_version_file }}"
        file_system: "bootflash:"
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: 10
        delay_factor: 10
    - name: "1b. Test stringed integer"
      ntc_file_copy:
        platform: cisco_ios_ssh
        local_file: /home/ios-images/4500/{{ approved_version_file }}
        remote_file: "{{ approved_version_file }}"
        file_system: "bootflash:"
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: '10'
        delay_factor: '10'
      ignore_errors: yes
    - name: "1c. Test string"
      ntc_file_copy:
        platform: cisco_ios_ssh
        local_file: /home/ios-images/4500/{{ approved_version_file }}
        remote_file: "{{ approved_version_file }}"
        file_system: "bootflash:"
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: 'ten'
        delay_factor: 'ten'
      ignore_errors: yes`

Test 1a, 1b runs successfully.

Test 1c fails with the following message:
"msg": "argument global_delay_factor is of type <type 'str'> and we were unable to convert to int: invalid literal for int() with base 10: 'ten'"


`  tasks:
    - name: "1a. Test with an integer"
      ntc_save_config:
        platform: cisco_ios_ssh
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: 10
        delay_factor: 10
      ignore_errors: yes
    - name: "1b. Test stringed integer"
      ntc_save_config:
        platform: cisco_ios_ssh
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: '10'
        delay_factor: '10'
      ignore_errors: yes
    - name: "1c. Test string"
      ntc_save_config:
        platform: cisco_ios_ssh
        host: "{{ inventory_hostname }}"
        username: username
        password: "{{ ansible_ssh_pass }}"
        global_delay_factor: 'ten'
        delay_factor: 'ten'
      ignore_errors: yes`

Test 1a, 1b runs successfully.

Test 1c fails with the following message:
"msg": "argument global_delay_factor is of type <type 'str'> and we were unable to convert to int: invalid literal for int() with base 10: 'ten'"

**Pull request merged**